api/v2/routes_database.go

```diff
@@ -21,20 +23,31 @@ func (api *API) searchUploadsForUser(c *gin.Context) {
 		FailWithMissingField(c, missingField)
 		return
 	}
-	// escape string
+	// escape string to prevent html encoded characters causing issues
 	forms["search_query"] = html.UnescapeString(forms["search_query"])
+	// force lower-case to make matching more likely s
+	lower := strings.ToLower(forms["search_query"])
+	// just in case lets try and avoid any possible headaches
+	if strings.Contains(lower, "drop") || strings.Contains(lower, "select") {
+		if strings.Contains(lower, "table") ||
+			strings.Contains(lower, "column") ||
+			strings.Contains(lower, "row") {
+			Fail(c, errors.New("possible sql injection attack, goodbye"), http.StatusBadRequest)
+			return
+		}
```

Comment:
So technically the way we are using GORM isn't vulnerable to injection attacks, however there has been cases in the past where gorm has a bug that can introduce injection attacks. Additionally I would much rather be extra safe, because this API call allows arbitrary user input.

If we get sufficient complaints we can remove this check, but I think there's no harm using this for now. I have slightly adjusted the matching to be less likely to get a false positive.